### PR TITLE
[Gateway] HTTP server

### PR DESCRIPTION
**Pull request by bolekk**
Adding
1. HTTP server for user requests.
2. ProcessRequest function that decodes the request, calls a handler and encodes the response.
3. Script to run Gateway outside of the core node, for testing purposes.

### CODE REVIEW

File: core/services/gateway/network/httpserver.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+//go:generate mockery --quiet --name HttpServer --output ./mocks/ --case=underscore
+type HttpServer interface {
+	job.ServiceCtx
+
+	SetHTTPRequestHandler(handler HTTPRequestHandler)
```
Comment (pinebit) [inline core/services/gateway/network/httpserver.go:19]: Have a comment: not thread-safe.

File: core/services/gateway/network/httpserver.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	WriteTimeoutMillis   uint32
+	RequestTimeoutMillis uint32
+}
+
+type httpServer struct {
+	utils.StartStopOnce
+	config  *HTTPServerConfig
+	server  *http.Server
+	handler HTTPRequestHandler
+	done    chan bool
```
Comment (pinebit) [inline core/services/gateway/network/httpserver.go:44]: Core naming pattern is `xyzCh` (Ch suffix) for any channel.

File: core/services/gateway/handler.go [hunk 3]
```diff
@@ -23,13 +23,15 @@ type UserCallbackPayload struct {
...
+//go:generate mockery --quiet --name Handler --output ./mocks/ --case=underscore
+
 type Handler interface {
 	job.ServiceCtx
 
 	// Each user request is processed by a separate goroutine, which:
 	//   1. calls HandleUserMessage
 	//   2. waits on callbackChan with a timeout
-	HandleUserMessage(ctx context.Context, msg *Message, callbackChan chan UserCallbackPayload) error
+	HandleUserMessage(ctx context.Context, msg *Message, callbackChan chan<- UserCallbackPayload) error
```
Comment (pinebit) [inline core/services/gateway/handler.go:34]: Core naming pattern is `xyzCh` (Ch suffix) for any channel.

**Review by pinebit — CHANGES_REQUESTED** [review]
Provided my comments.

File: core/services/gateway/network/httpserver.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+type HTTPServerConfig struct {
+	Host                 string
+	Port                 uint16
+	TLSEnabled           bool
+	TLSCertPath          string
+	TLSKeyPath           string
+	Path                 string
+	ContentTypeHeader    string
+	ReadTimeoutMillis    uint32
```
[the remark this answers is not in the document]
Comment (bolekk) [reply core/services/gateway/network/httpserver.go:34]: It looks like I would need to use the Duration struct from core/store/models.go or implement something similar myslef. I'm not too keen on adding another core dependency so I'll leave it for now.

**Review by pinebit — APPROVED** [review]

**Pull request merged**